src/dcos_e2e/docker_utils/__init__.py

```diff
@@ -0,0 +1,122 @@
+from typing import Any, Dict, Optional, Tuple
+
+import docker
+
+
+class DockerLoopbackVolume():
+    """
+    A loopback device sidecar, created in a Docker container.
+    """
+
+    def __init__(
+        self,
+        size: int,
+        labels: Optional[Dict[str, str]] = None,
+    ) -> None:
+        """
+        Create a loopback device pointing to a block device running
+        in a container. This can be used to provide an unformatted
+        disk to a cluster.
+
+        Args:
+            size: Size of the block device in Megabytes.
+            labels: Docker labels to add to the container.
+        """
+        self.size = size
+        docker_container_labels = labels or {}
+
+        self.container, self.path = \
+            DockerLoopbackVolume.create(size, docker_container_labels)
+
+    @staticmethod
+    def create(
+        size: int,
+        labels: Optional[Dict[str, str]] = None,
+    ) -> Tuple[docker.models.containers.Container, str]:
+        """
+        Create a loopback device pointing to a block device running
+        in a container.
+
+        Args:
+            size: Size of the block device in Megabytes.
+            labels: Docker labels to add to the container.
+        Returns:
+            A tuple containing the container and the path of
+            the loopback device.
+        """
+        client = docker.from_env(version='auto')
+
+        container = client.containers.create(
+            name='dcos-e2e-loopback-sidecar',
+            privileged=True,
+            detach=True,
+            tty=True,
+            image='centos:7',
```

Comment:
Not complete thought:

IIRC we discussed: Does this pull the image? Why CentOS 7? Is this pulled by `make pull-images` (so the network access in CI is minimised during test run time)?

Reply:
Added a comment on why CentOS 7 is chosen here.